posh-dotnet.psm1

```diff
@@ -50,6 +50,37 @@ filter script:MatchingCommand($commandName)
     }
 }
 
+# extracts help for dotnet cli v2
+function Get-HelpTextV2HashTable
+{
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingInvokeExpression", "")]
+    Param
+    (
+        $commandAst,
+        $completionList
+    )
+
+    $help = Invoke-Expression "$([string]::Join(" ", $commandAst.CommandElements)) --help"
```

Comment:
This approach has some risk.
If the original command line used a variable, e.g.
```
dotnet run $completionList <TAB>
```

Then `$completionList` is looked up in the current scope, finding your definition, and I'm guessing bad things could happen.

The bottom line is - `Invoke-Expression` can be dangerous if you aren't validating the input.

Reply:
Thanks for your review. I was aware that `iex` has a bad reputation but do you have any thoughts on an alternative? I could manually code 4 if statements for the case  of 0,1,2,3 arguments where I would use the `&` operator then. In terms of validation, I could check that the first command is `dotnet` and that no `CommandElement` contains a semicolon. What are your thoughts?

Reply:
Because you're working from the Ast, you're already somewhat immune to things like a semicolon, but you might find helpers in `System.Management.Automation.Language.CodeGeneration` useful.

I don't think there can be anything funny in the command name or tab completion wouldn't route to your code, but I suppose future improvements could change that, e.g. in the future something like this might work:

```powershell
$cmd = 'dotnet'
& $cmd <TAB>
```

I guess if each element is a `System.Management.Automation.Language.StringConstantExpressionAst` then that should be enough validation.

Reply:
Thanks for your advice and insider knowledge sharing :-)
I made it strongly typed [here](https://github.com/bergmeister/posh-dotnet/blob/master/posh-dotnet.psm1#L60) to mitigate the risk and played a bit around with all sorts of hacks but could not manage to inject any command.